pkg/config/generate_test.go

```diff
@@ -134,3 +134,40 @@ func testRequiredFields(cs *api.OpenShiftManagedCluster, t *testing.T) {
 	assert(c.NodeBootstrapKubeconfig != nil, "NodeBootstrapKubeconfig")
 	assert(c.AzureClusterReaderKubeconfig != nil, "AzureClusterReaderKubeconfig")
 }
+
+func TestInvalidateSecrets(t *testing.T) {
+	dpc := &api.PluginConfig{}
+	populate.Walk(dpc, nil)
+	cs := &api.OpenShiftManagedCluster{}
+	populate.Walk(cs, nil)
+
+	g := &simpleGenerator{
+		pluginConfig: *dpc,
+	}
+	saved := *cs
+	if err := g.InvalidateSecrets(cs); err != nil {
+		t.Errorf("configGenerator.InvalidateSecrets error = %v", err)
+	}
+	if !reflect.DeepEqual(saved.Config, cs.Config) {
```

Comment:
This test feels pretty weak.  @kargakis @mjudeikis @asalkeld do you have any thoughts on how to strengthen it in a future proof way?

Reply:
First compare all fields that are expected to be different then assign `saved` values back to those changed in `cs`, then expect `reflect.DeepEqual(saved.Config, cs.Config) == true`?

Reply:
I think we need to make sure we match all the changes and not changed values too. I think something under these lines 
as it was done here https://github.com/openshift/openshift-azure/blob/a22eecf16d8cb38d7bca600ef2de02143f3addba/pkg/api/types_test.go#L368 ?